src/buchheim.jl

```diff
@@ -35,17 +49,34 @@ function Tree(tree::AbstractVector, nodesize)
     change = zeros(len)
     ancestor = collect(1:len)
     nodes = copy(tree)
-    positions = zeros(Point{2,Float64}, len)
+    positions = zeros(Point{2,Ptype}, len)
     t = Tree(nodes, mod, thread, ancestor, prelim, shift, change, positions, nodesize)
     return t
 end
 
-function layout(t::AbstractVector; nodesize=ones(length(t)))
-    return layout!(t, nodesize)
+function adj_mat_to_list(M::AbstractMatrix)
+    N = size(M, 1)
+    list = Vector{Vector{Int}}(undef, N)
+    for i in 1:N
+        list[i] = findall(x -> !iszero(x), view(M, i, :))
+    end
+    return list
 end
 
-function layout!(t::AbstractVector, nodesize)
-    tree = Tree(t, nodesize)
+function layout(para::Buchheim, adj_matrix::AbstractMatrix)
+    @assert size(adj_matrix, 1) == size(adj_matrix, 2) "adjacency matrix not square!"
```

Comment:
We will do something like this for every layout anyway? Maybe then we can have a `assert_square` function, that throws  some kind of error (assertions with the `@assert` macro might be disabled under some circumstances in the future)